src/local-editor-service.ts

```diff
@@ -247,19 +270,7 @@ export class LocalEditorService extends EditorService {
   }
 
   async getWarningsForFile(localPath: string): Promise<Warning[]> {
-    try {
-      const doc = await this._analyzer.analyze(localPath);
-      return doc.getWarnings();
-    } catch (e) {
-      // This might happen if, e.g. `localPath` has a parse error. In that case
-      // we can't construct a valid Document, but we might still be able to give
-      // a reasonable warning.
-      if (e instanceof WarningCarryingException) {
-        const warnException: WarningCarryingException = e;
-        return [warnException.warning];
-      }
-      throw e;
-    }
+    return this._linter.lint([localPath]);
```

Comment:
should we update the interface of LocalEditorService as well, to use a public `lint` function instead of more outdated `getWarningForFile`?